job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java

```diff
@@ -46,27 +46,27 @@
   private final CsvSchema mSchema;
 
   /**
-   * @param fs the hdfs compatible client
    * @param inputPath the input path
    * @throws IOException when failed to open the input
    */
-  private CsvReader(FileSystem fs, Path inputPath) throws IOException {
+  private CsvReader(Path inputPath) throws IOException {
     mCloser = Closer.create();
-    mFs = fs;
+    Configuration conf = ReadWriterUtils.readNoCacheConf();
+    mFs = inputPath.getFileSystem(conf);
     mCloser.register(mFs);
     CSVProperties props = new CSVProperties.Builder()
         .hasHeader()
         .build();
     Schema schema;
-    try (InputStream inputStream = open(fs, inputPath)) {
+    try (InputStream inputStream = open(mFs, inputPath)) {
       String schemaName = inputPath.getName();
       if (schemaName.contains(".")) {
         schemaName = schemaName.substring(0, schemaName.indexOf("."));
       }
       schema = AvroCSV.inferSchema(schemaName, inputStream, props);
       mSchema = new CsvSchema(schema);
     }
-    mReader = new AvroCSVReader<>(open(fs, inputPath), props, schema, Record.class, false);
+    mReader = new AvroCSVReader<>(open(mFs, inputPath), props, schema, Record.class, false);
```

Comment:
If we throw an exception in the constructor should we make sure to close the closer?